src/Server.php

```diff
@@ -167,7 +167,13 @@ public function listen(callable $finalHandler = null)
         if (! $response instanceof ResponseInterface) {
             $response = $this->response;
         }
-        $this->getEmitter()->emit($response, $bufferLevel);
+
+        $emitter = $this->getEmitter();
+        if ($emitter instanceof Response\SapiEmitter) {
+            $emitter->emit($response, $bufferLevel);
+        } else {
+            $emitter->emit($response);
+        }
```

Comment:
This really breaks basic OO principles for the sake of backward compatibility.
IMHO, #233 is better, even if not backward compatible (The `SapiStreamEmmiter` is almost never consumed by anyone else but Diactoros Server class)